src/App/Fossa/YamlDeps.hs

```diff
@@ -0,0 +1,196 @@
+import Data.Foldable (traverse_)
+import Data.List.NonEmpty qualified as NE
+import Data.String.Conversion (toText, toString)
+import Data.Text (Text, unpack)
+import DepTypes (DepType(..))
+import Effect.ReadFS (ReadFS, doesFileExist, readContentsYaml)
+import Path
+import Srclib.Converter (depTypeToFetcher)
+import Srclib.Types (AdditionalDepData (..), Locator (..), SourceUnit (..), SourceUnitBuild (..), SourceUnitDependency (SourceUnitDependency), SourceUserDefDep (..))
+import Data.HashMap.Strict (member)
+import Control.Monad (when)
+
+analyzeFossaDepsYaml :: (Has Diagnostics sig m, Has ReadFS sig m) => Path Abs Dir -> m (Maybe SourceUnit)
+analyzeFossaDepsYaml root = do
+  maybeDepsFile <- findFossaDepsFile root
+  case maybeDepsFile of
+    Nothing -> pure Nothing
+    Just depsFile -> do
+      yamldeps <- context "Reading fossa-deps file" $ readContentsYaml depsFile
+      context "Converting fossa-deps to partial API payload" $ toSourceUnit root yamldeps
+
+findFossaDepsFile :: (Has Diagnostics sig m, Has ReadFS sig m) => Path Abs Dir -> m (Maybe (Path Abs File))
+findFossaDepsFile root = do
+  let ymlFile = root </> $(mkRelFile "fossa-deps.yml")
+      yamlFile = root </> $(mkRelFile "fossa-deps.yaml")
+  ymlExists <- doesFileExist ymlFile
+  yamlExists <- doesFileExist yamlFile
+  case (ymlExists, yamlExists) of
+    (True, True) -> fatalText "Found '.yml' and '.yaml' files when searching for fossa-deps, only one is permitted if present."
+    (True, False) -> pure $ Just ymlFile
+    (False, True) -> pure $ Just yamlFile
+    (False, False) -> pure Nothing
+
+toSourceUnit :: Has Diagnostics sig m => Path Abs Dir -> YamlDependencies -> m (Maybe SourceUnit)
+toSourceUnit root yamldeps@YamlDependencies{..} = do
+  when (hasNoDeps yamldeps) $ fatalText "No dependencies found in fossa-deps file"
+  let renderedPath = toText root
+      build = toBuildData <$> NE.nonEmpty referencedDependencies
+      additional = toAdditionalData <$> NE.nonEmpty customDependencies
+  pure . Just $
+    SourceUnit
+      { sourceUnitName = renderedPath,
+        sourceUnitManifest = renderedPath,
+        sourceUnitType = "user-specific-yaml",
+        sourceUnitBuild = build,
+        additionalData = additional
+      }
+
+toBuildData :: NE.NonEmpty ReferencedDependency -> SourceUnitBuild
+toBuildData deps =
+  SourceUnitBuild
+    { buildArtifact = "default",
+      buildSucceeded = True,
+      buildImports = imports,
+      buildDependencies = buildDeps
+    }
+  where
+    (imports, buildDeps) = unzip . map (addEmptyDep . toImport) $ NE.toList deps
+
+    toImport :: ReferencedDependency -> Locator
+    toImport ReferencedDependency {..} =
+      Locator
+        { locatorFetcher = depTypeToFetcher locDepType,
+          locatorProject = locDepName,
+          locatorRevision = locDepVersion
+        }
+
+    addEmptyDep :: Locator -> (Locator, SourceUnitDependency)
+    addEmptyDep loc = (loc, SourceUnitDependency loc [])
```

Comment:
I think this might be clearer without the `unzip`

```suggestion
      buildImports = imports,
      buildDependencies = map addEmptyDep imports
    }
  where
    imports = map toImport $ NE.toList deps
    
    addEmptyDep :: Locator -> SourceUnitDependency
    addEmptyDep loc = SourceUnitDependency loc []
```